docs/source/55_api.md

```diff
@@ -339,6 +340,138 @@ val bcastFeatures = sc.loadFeatures("my/features.adam").broadcast()
+could perform this query to extract all genotypes that fall in exonic regions.
+
+```scala
+// Inner join will filter out genotypic data not represented in the feature dataset
+val genotypes = sc.loadGenotypes(“my/genotypes.adam”)
+val features = sc.loadFeatures(“my/features.adam”)
+
+// We can use ShuffleRegionJoin…
+val joinedGenotypesShuffle = genotypes.shuffleRegionJoin(features)
+
+// …or BroadcastRegionJoin
+val joinedGenotypesBcast = features.broadcastRegionJoin(genotypes)
+
+// In the case that we only want Genotypes, we can use a simple predicate
+val filteredGenotypesShuffle = joinedGenotypesShuffle.rdd.map(_._1)
+
+val filteredGenotypesBcast = joinedGenotypesBcast.rdd.map(_._2)
+```
+
+After the join, we can perform a predicate function on the resulting RDD to
+manipulate it into providing the answer to our question. Because we were
+interested in the Genotypes that overlap the features, we used a map function
+to extract them.
+
+Another important distinction between ShuffleRegionJoin and BroadcastRegionJoin
+is that in a BroadcastRegionJoin, the left dataset is sent to all executors. In
+this case, we chose to send the `features` dataset because feature data is
+usually smaller in size than genotypic data.
+
+###### Group overlapping variant data by the gene they overlap
+
+This query joins an RDD of Variants against an RDD of Features, and immediately
+performs a group-by on the Feature. This produces an RDD whose elements are a
+tuple containing a Feature, and all of the Variants overlapping the Feature.
+This query is useful for trying to identify annotated variants that may
+interact (identifying frameshift mutations within a transcript that may act as
+a pair to shift and then restore the reading frame) or as the start of a query
+that computes variant density over a set of genomic features.
+
+```scala
+// Inner join with a group by on the features
+val features = sc.loadFeatures(“my/features.adam”)
+val variants = sc.loadVariants(“my/variants.adam")
+
+// As a ShuffleRegionJoin, it can be implemented as follows:
+val variantsByFeatureShuffle = features.shuffleRegionJoinAndGroupByLeft(variants)
+
+// As a BroadcastRegionJoin, it can be implemented as follows:
+val variantsByFeatureBcast = variants.broadcastRegionJoinAndGroupByRight(features)
+```
+
+When we switch join strategies, we need to change the dataset that is on the
+left side of the join. This distinction is very important to understanding the
+difference between BroadcastRegionJoin and ShuffleRegionJoin.
+
+To perform a `groupBy` after the join, BroadcastRegionJoin only supports
+grouping by the right dataset, and ShuffleRegionJoin supports only grouping by
+the left dataset. Thus, depending on the type of join, the left and right
+dataset may change between BroadcastRegionJoin and ShuffleRegionJoin.
+
+The reason BroadcastRegionJoin does not have a `joinAndGroupByLeft`
+implementation is due to the fact that the left dataset is broadcasted to all
```

Comment:
Seems reasonable. Might tighten up to:

> Unlike shuffle joins, broadcast joins don't maintain a sort order invariant. Because of this, we would need to shuffle all data to a group-by on the left side of the dataset, and there is no opportunity to optimize by combining the join and group-by.